libvirt/resource_libvirt_domain.go

```diff
@@ -106,6 +118,34 @@ func resourceLibvirtDomainCreate(d *schema.ResourceData, meta interface{}) error
 		domainDef.Metadata.TerraformLibvirt.Xml = metadata.(string)
 	}
 
+	if firmware, ok := d.GetOk("firmware"); ok {
+		firmwareFile, err := homedir.Expand(firmware.(string))
```

Comment:
I think we can remove it, this is something that is not done by any program I know. Also the user can place these files into the project directory and then terraform will automatically find them (`firmware = "foo"` -> `<pwd>/foo`)